include/drivers/gpio.h

```diff
@@ -405,7 +405,18 @@ typedef u32_t gpio_port_value_t;
  * element in the driver's struct driver_data decleration.
  */
 struct gpio_driver_data {
+	/* Mask identifying pins that are configured as active low.
+	 *
+	 * Management of this mask is the responsibility of the
+	 * wrapper functions in this header.
+	 */
 	gpio_port_pins_t invert;
+
+	/* Number of pins supported by the specific driver.
+	 *
+	 * This must be set by the driver during device initialization.
+	 */
+	u8_t num_pins;
```

Comment:
Number of pins can be determined during the compile time and is constant. We could put it inside `dev->config->config_info` structure? That saves 4 bytes of RAM. It is however more work since we currently don't expose `dev->config->config_info` at the API level.

Reply:
We'd also have to define a common structure that everybody would have to put at the start of their private config structure.   Since we already do that for data, that's not too far a stretch, but it touches every driver.

It also can't be done until the work started in #20034 is [completed](https://github.com/zephyrproject-rtos/zephyr/pull/20034#discussion_r338619559).

Reply:
Moved to common config structure, some drivers updated.